src/library_async.js

```diff
@@ -340,6 +340,7 @@ mergeInto(LibraryManager.library, {
     });
   },
 
+  emscripten_scan_registers__deps: ['$getDynCaller'],
```

Comment:
Why is this dep added? I don't see the function used in the body. I'd guess it's expanded from `{{{ makeDynCall }}}`? It seems confusing to not see the function in the dep actually used, and could make us forget deps later on. Could the one in the body be changed?

Reply:
Yes, any function with a `makeDynCall` need to depend on `getDynCaller` for now.

`getDynCaller` then selects between the legacy native thunks and the direct table access.   Once we can remove the legacy thunks completely we can look at making this more efficient.